src/pages/Opening/Opening.jsx

```diff
@@ -0,0 +1,246 @@
+                    return (
+                      <TableRow key={openingId}>
+                        <TableColumn>{opening.jobTitle}</TableColumn>
+                        <TableColumn>{opening.jobDescription}</TableColumn>
+                        <TableColumn>{opening.company}</TableColumn>
+                        <TableColumn>
+                          <NumberFormat value={opening.maxSalaryRange} displayType={'text'} thousandSeparator={true} prefix={'$'} />
+                        </TableColumn>
+                        <TableColumn>{opening.status}</TableColumn>
+                        <MenuButtonColumn
+                          icon
+                          menuItems={this.menuItems.map(m =>
+                            m.key === 'edit'
+                              ? { ...m, onClick: () => this.onClickEdit(opening) }
+                              : { ...m, onClick: () => this.onClickDelete(opening) }
+                          )}
+                          listClassName="action-menu"
+                        >
+                          <FontIcon>more_vert</FontIcon>
+                        </MenuButtonColumn>
+                      </TableRow>
+                    );
+                  })}
+                </TableBody>
+              </DataTable>
+            );
+          }}
+        </Query>
+        <Mutation
+          mutation={!this.opening.id ? ADD_OPENING : UPDATE_OPENING}
+          onCompleted={this.onSaveSuccess}
+          onError={this.onSaveError}
+          refetchQueries={[{ query: OPENING_QUERY }]}
+          awaitRefetchQueries={true}
+        >
+          {createUpdateOpening => (
+            <OpeningDialog
+              showDialog={this.state.showOpeningDialog}
+              openingToEdit={this.opening}
+              hideModal={this.hideDialog}
+              createUpdateMutation={createUpdateOpening}
+            />
+          )}
+        </Mutation>
+        <Mutation
+          mutation={DELETE_OPENING}
+          onCompleted={this.onDeleteSuccess}
+          refetchQueries={[{ query: OPENING_QUERY }]}
+          awaitRefetchQueries={true}
+        >
+          {deleteOpening => (
+            <ConfirmDialog
+              visible={alertVisible}
+              title="Alert"
+              message="Do you want to delete this opening?"
+              onEvent={e => this.onDeleteEvent(e, deleteOpening)}
+            />
+          )}
+        </Mutation>
+        <Button floating primary className="bottom-right" onClick={this.showDialog}>
+          <FontIcon>add</FontIcon>
+        </Button>
+        <Snackbar toasts={toasts} autohide={true} onDismiss={this.dismissToast} />
+      </Fragment>
+    );
+  }
+}
+
+const OPENING_QUERY = gql`
+  {
+    openings {
+      id
+      jobTitle
+      jobDescription
+      company
+      maxSalaryRange
+      status
+      steps {
+        id
+      }
+      applications {
+        id
+      }
+    }
+  }
+`;
+
+const ADD_OPENING = gql`
```

Comment:
there is an `upsertOpening` mutation that can be used for adding and updating at the same time

Reply:
Done